### PR TITLE
# feat(): updated fxa nonprod queries updated to be in line with production queries

### PR DESCRIPTION
**Pull request by kik-kik**
# feat(): updated fxa nonprod queries updated to be in line with production queries

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-1562)

### CODE REVIEW

**Review by akkomar — APPROVED** [review]
LGTM, I'd consider if we even need these.

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stderr_events_v1/query.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  SPLIT(jsonPayload.logger, "-")[OFFSET(1)] AS fxa_server,
+  * REPLACE (
+    (
+      SELECT AS STRUCT
+        jsonPayload.* REPLACE (
+          (
+            SELECT AS STRUCT
+              jsonPayload.fields.* EXCEPT (device_id, user_id),
+              TO_HEX(SHA256(jsonPayload.fields.user_id)) AS user_id,
+              TO_HEX(SHA256(jsonPayload.fields.device_id)) AS device_id
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stderr_events_v1/query.sql:12]: Suggest using `REPLACE ()` instead so `user_id` and `device_id` aren't always pushed to the end.
```suggestion
            SELECT AS STRUCT
              jsonPayload.fields.* REPLACE (
                TO_HEX(SHA256(jsonPayload.fields.user_id)) AS user_id,
                TO_HEX(SHA256(jsonPayload.fields.device_id)) AS device_id
              )
```
(ditto in `nonprod_fxa_gcp_stdout_events_v1`)

File: sql/moz-fx-data-shared-prod/firefox_accounts/nonprod_fxa_all_events/view.sql [hunk 2]
```diff
@@ -69,13 +69,33 @@ stdout_events AS (
...
+gcp_stdout_events AS (
   SELECT
     fxa_server,
     `timestamp`,
     receiveTimestamp,
-    TIMESTAMP_MILLIS(CAST(jsonPayload.fields.time AS INT64)) AS event_time,
+    SAFE.TIMESTAMP_MILLIS(SAFE_CAST(jsonPayload.fields.time AS INT64)) AS event_time,
+    jsonPayload.fields.user_id,
+    CAST(NULL AS STRING) AS country,
+    CAST(NULL AS STRING) AS country_code,
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/firefox_accounts/nonprod_fxa_all_events/view.sql:80]:
```suggestion
    jsonPayload.fields.country,
    JSON_VALUE(jsonPayload.fields.event_properties, "$.country_code") AS country_code,
```

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stdout_events_v1/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+friendly_name: Non-Prod FxA Stdout Events
+description: |-
+  Staging table containing stderr fxa logs
```
[the remark this answers is not in the document]
Comment (sean-rose) [reply sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stdout_events_v1/metadata.yaml:3]: The code suggestion here still applies.

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stderr_events_v1/query.sql (hunk 1 above)
Comment (sean-rose) [reply sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stderr_events_v1/query.sql]: The same still needs to be done in `nonprod_fxa_gcp_stdout_events_v1`.

**Review by sean-rose — APPROVED** [review]
r+wc

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stderr_events_v1/query.sql (hunk 1 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stderr_events_v1/query.sql]: ok, I made the tweaks @sean-rose. I hope now everything looks in order.

**Pull request merged**